File: Command/Java/src/Commands/HottubOnCommand.java
package Commands;

import Devices.Hottub;

public class HottubOnCommand implements Command {
    Hottub hottub;

    public HottubOnCommand(Hottub hottub) {
        this.hottub = hottub;
    }
    public void execute() {
        hottub.on();
        hottub.setTemperature(40);
        hottub.circulate();
    }
    public void undo() {
        hottub.off();
    }
}

File: Command/TypeScript/src/commands/HottubOnCommand.ts
import Hottub from '../devices/Hottub';
import Command from './Command';

export default class HottubOnCommand implements Command {
  hottub: Hottub;

  constructor(hottub: Hottub) {
    this.hottub = hottub;
  }

  execute() {
    this.hottub.setTemperature(40);
    this.hottub.on();
  }

  undo() {
    this.hottub.off();
  }
}
